src/test/java/io/confluent/kafkarest/unit/ConsumerManagerTest.java

```diff
@@ -172,38 +177,50 @@ public void testConsumerNormalOps() throws InterruptedException, ExecutionExcept
     String cid = consumerManager.createConsumer(
         groupName, new ConsumerInstanceConfig(EmbeddedFormat.BINARY));
     sawCallback = false;
+    actualException = null;
```

Comment:
@ewencp I've taken a quick look  at the implementation in Kafka of ThreadedTest.

In summary, I think we would need:

- I'll need to create a new abstract class that extends Thread and includes a static UncaughtExceptionHandler.
- ConsumerWorker will need to then extend that new class
- Tests will have to extend ThreadedTest class to swap in the uncaught exception handler for the tests

I did take a look if I could add a few references and then use the ShutdownableThread class from Kafka but that hooks in a bunch of other stuff that I don't think is needed.

So I guess the question becomes - do you want me to do the work to insert that abstract class OR are you happy to go with the tests as is?

Ta,
Alan